framework/include/utils/InputParameterWarehouse.h

```diff
@@ -127,4 +136,61 @@ class InputParameterWarehouse : public Warehouse<InputParameters>
   friend class SolidMaterialProperties;
 };
 
+
+template<typename T>
+ControllableParameter<T>
+InputParameterWarehouse::getControllableParameter(const MooseObjectParameterName & input, bool mark_as_controlled /*=false*/)
+{
+
+  // The ControllableParameter object to return
+  ControllableParameter<T> output;
+
+  // Vector of desired parameters
+  std::vector<MooseObjectParameterName> params(1, input);
+  std::map<MooseObjectParameterName, std::vector<MooseObjectParameterName> >::const_iterator link_it = _input_parameter_links.find(input);
+  if (link_it != _input_parameter_links.end())
+    params.insert(params.end(), link_it->second.begin(), link_it->second.end());
+
+  // Parameter object iterator
+  std::multimap<MooseObjectName, MooseSharedPointer<InputParameters> >::const_iterator iter;
+
+  // Loop over all threads
+  for (THREAD_ID tid = 0; tid < libMesh::n_threads(); ++tid)
+  {
+    // Loop over all InputParameter objects
+    for (iter = _input_parameters[tid].begin(); iter != _input_parameters[tid].end(); ++iter)
+    {
+      // Loop of all desired params
+      for (std::vector<MooseObjectParameterName>::iterator it = params.begin(); it != params.end(); ++it)
```

Comment:
Agreed this functions is a bit "wonky", let me explain what it is doing.

(1) First a vector of "desired" params is built, which includes the parameter of interest and any parameters it may be linked to, this is the `params` vector.
(2) Controls are not execute in a threaded region, but we need to change values on all the threads. Thus, the loop of tid.
(3) The next loop is over the `InputParameters` objects (not individual parameters), this is just a linear search, but since we are looking for parameter(s) there would need to be some sort of map of parameters to InputParameters, which is a bit overkill, at least for now.
(4) The next loop checks if any of the "desired" parameters exist on the current InputParameters object, if it does and is allowed to be controlled insert it into the `ControllableParameter` object to be output.